simulation/traffic_simulator/src/behavior/longitudinal_speed_planning.cpp

```diff
@@ -0,0 +1,326 @@
+    } else {
+      return (twist.linear.x - target_speed) / constraints.max_deceleration;
+    }
+  }
+}
+
+auto LongitudinalSpeedPlanner::isReachedToTargetSpeedWithConstantJerk(
+  double target_speed, const traffic_simulator_msgs::msg::DynamicConstraints & constraints,
+  const geometry_msgs::msg::Twist & current_twist, const geometry_msgs::msg::Accel & current_accel,
+  double duration, double tolerance) const -> bool
+{
+  if (
+    std::abs(
+      getVelocityWithConstantJerk(
+        target_speed, current_twist, current_accel, constraints, duration) -
+      target_speed) <= std::abs(tolerance)) {
+    return true;
+  }
+  return false;
+}
+
+auto LongitudinalSpeedPlanner::isAccelerating(
+  double target_speed, const geometry_msgs::msg::Twist & current_twist) const -> bool
+{
+  if (current_twist.linear.x > target_speed) {
+    return false;
+  }
+  return true;
+}
+
+auto LongitudinalSpeedPlanner::isDecelerating(
+  double target_speed, const geometry_msgs::msg::Twist & current_twist) const -> bool
+{
+  if (current_twist.linear.x < target_speed) {
+    return false;
+  }
+  return true;
+}
+
+auto LongitudinalSpeedPlanner::planLinearJerk(
+  double target_speed, const traffic_simulator_msgs::msg::DynamicConstraints & constraints,
+  const geometry_msgs::msg::Twist & current_twist,
+  const geometry_msgs::msg::Accel & current_accel) const -> double
+{
+  double accel_x_new = 0;
+  if (isAccelerating(target_speed, current_twist)) {
+    accel_x_new = boost::algorithm::clamp(
+      current_accel.linear.x + step_time * constraints.max_acceleration_rate, 0,
+      std::min(constraints.max_acceleration, (target_speed - current_twist.linear.x) / step_time));
+  } else {
+    accel_x_new = boost::algorithm::clamp(
+      current_accel.linear.x - step_time * constraints.max_deceleration_rate,
+      std::max(
+        constraints.max_deceleration * -1, (target_speed - current_twist.linear.x) / step_time),
+      0);
+  }
+  return (accel_x_new - current_accel.linear.x) / step_time;
+}
+
+auto LongitudinalSpeedPlanner::forward(
+  double linear_jerk, const geometry_msgs::msg::Accel & accel,
+  const traffic_simulator_msgs::msg::DynamicConstraints & constraints) const
+  -> geometry_msgs::msg::Accel
+{
+  geometry_msgs::msg::Accel ret = accel;
+  ret.linear.x = accel.linear.x + step_time * linear_jerk;
+  ret.linear.x = boost::algorithm::clamp(
+    ret.linear.x, constraints.max_deceleration * -1, constraints.max_acceleration);
+  return ret;
+}
+
+auto LongitudinalSpeedPlanner::forward(
+  const geometry_msgs::msg::Accel & accel, const geometry_msgs::msg::Twist & twist,
+  const traffic_simulator_msgs::msg::DynamicConstraints & constraints) const
+  -> geometry_msgs::msg::Twist
+{
+  geometry_msgs::msg::Twist ret = twist;
+  ret.linear = ret.linear + accel.linear * step_time;
+  ret.linear.x =
+    boost::algorithm::clamp(ret.linear.x, -1 * constraints.max_speed, constraints.max_speed);
```

Comment:
boost::algorithm::clamp => std::clamp

Reply:
remove boost::algorithm::clamp and use std::clamp